internal/fuzzbinder/fuzzscraper.go

```diff
@@ -0,0 +1,265 @@
+// statistics from the ClusterFuzz and OSS-Fuzz reports.
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"strconv"
+	"strings"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+// CoverageBucket is the OSS-Fuzz bucket containing the coverage reports.
+const CoverageBucket = "oss-fuzz-coverage"
+
+// CoverageSummary contains the coverage report summary.
+type CoverageSummary struct {
+	Data    []SummaryData `json:"data"`
+	Type    interface{}   `json:"type"`
+	Version interface{}   `json:"version"`
+}
+
+// SummaryData contains the data of the coverage report summary.
+type SummaryData struct {
+	Files  interface{}                     `json:"files"`
+	Totals map[string](map[string]float64) `json:"totals"`
+}
+
+// getBucket gets a GCS bucket given its name.
+func getBucket(bucketName string) (*storage.BucketHandle, error) {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	bucket := client.Bucket(bucketName)
+	return bucket, nil
+}
+
+// listBlobs gets all the objects in a GCS bucket under a given relative path.
+func listBlobs(bucket *storage.BucketHandle, relativePath string) ([]string, error) {
+	var blobs []string
+	ctx := context.Background()
+	query := &storage.Query{Prefix: relativePath}
+	objects := bucket.Objects(ctx, query)
+	for {
+		attrs, err := objects.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Bucket.Objects: %v", err)
+		}
+		blobs = append(blobs, attrs.Name)
+	}
+	return blobs, nil
+}
+
+// getBlob gets the file reader of a blob in a GCS bucket.
+func getBlob(bucket *storage.BucketHandle, blobName string) (*storage.Reader, error) {
+	ctx := context.Background()
+	reader, err := bucket.Object(blobName).NewReader(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Object(%q).NewReader: %v", blobName, err)
+	}
+	defer reader.Close()
+	return reader, nil
+}
+
+// getRev gets the revision of a source code given a source map file.
+func getRev(reader *storage.Reader, project string) (string, error) {
+	var payload map[string](map[string]string)
+	content, _ := io.ReadAll(reader)
+	err := json.Unmarshal(content, &payload)
+	if err != nil {
+		return "", fmt.Errorf("unmarshal: %v", err)
+	}
+	rev := payload[fmt.Sprintf("/src/%s", project)]["rev"]
+	return rev, nil
+}
+
+// parseCoverage gets the branch coverage and line coverage from a coverage report summary.
+func parseCoverage(reader *storage.Reader) (map[string]float64, map[string]float64, error) {
+	var payload CoverageSummary
+	content, _ := io.ReadAll(reader)
+	err := json.Unmarshal(content, &payload)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unmarshal: %v", err)
+	}
+	return payload.Data[0].Totals["branches"], payload.Data[0].Totals["lines"], nil
```

Comment:
I am not sure I understand why we take Data[0]? Are we sure the list is not empty? What about the remaining list elements?

Reply:
`payload.Data` is a list with one element. That's why I only get the first element. This is how the actual structure of the coverage summary reports saved in the coverage bucket. 
This list is not supposed to be empty from the examples I looked at, but I can verify that it is not empty in the code before using `payload.Data[0]` .